R/model.R

```diff
@@ -658,49 +724,80 @@ fall_run_model <- function(scenario = NULL, mode = c("seed", "simulate", "calibr
                                                  stochastic = stochastic)
 
       adults_in_ocean <- adults_in_ocean + ocean_entry_success
-
+      # # For use in the r2r metrics ---------------------------------------------
+      d <- data.frame(juveniles_at_chipps)
+      colnames(d) <- c("s", "m", "l", "vl")
+      d$watershed <- fallRunDSM::watershed_labels
+      d <- d |> tidyr::pivot_longer(names_to = "size",
+                                    values_to = "juveniles_at_chipps", -watershed)
+      d$year <- year
+      d$month <- month
+      output$juveniles_at_chipps <- dplyr::bind_rows(output$juveniles_at_chipps, d)
+      # end R2R metric -----------------------------------------------------------
     } # end month loop
-    d <- data.frame(juveniles_at_chipps)
-    colnames(d) <- c("s", "m", "l", "vl")
-    d$watershed <- fallRunDSM::watershed_labels
-    d <- d |> tidyr::pivot_longer(names_to = "size", values_to = "juveniles_at_chipps", -watershed)
-    d$year <- year
-    output$juveniles_at_chipps <- dplyr::bind_rows(output$juveniles_at_chipps, d)
-
     output$juvenile_biomass[ , year] <- juveniles_at_chipps %*% fallRunDSM::params$mass_by_size_class
-
-    adults_returning <- t(sapply(1:31, function(i) {
+    # Updated logic here for R2R so that natural adults and hatchery adults return separately
+    natural_adults_returning <- t(sapply(1:31, function(i) {
       if (stochastic) {
-        rmultinom(1, adults_in_ocean[i], prob = c(.25, .5, .25))
+        rmultinom(1, (adults_in_ocean[i]), prob = c(.25, .5, .25))
       } else {
-        round(adults_in_ocean[i] * c(.25, .5, .25))
+        round((adults_in_ocean[i])* c(.25, .5, .25))
       }
-    }))
+    })) * output$proportion_natural_juves_in_tribs[ , year]
+
+    # TODO see if better place to do this
+    natural_adults_returning[is.na(natural_adults_returning)] = 0
```

Comment:
i think you should switch to `is.nan`, NaN's are ok to appear in the code at this point (and be replaced) its just the places where denominator is 0, but if NA appears then something went wrong in the code you don't want to just replace those with zeros they should propagate and be fixed when they appear.

Reply:
updated